### ISSUE TITLE
Feature: change "Personal Workspace" title on personal workspace

### ISSUE THREAD

**Issue by isabensusan** [open-sauced/app#2811]
### Type of feature

🍕 Feature

### Current behavior

Personal workspaces by default are called "Personal Workspace", and it feels a bit confusing and detached for the user
<img width="1400" alt="image" src="https://github.com/open-sauced/app/assets/27322879/d5290c5d-8ed4-4741-95e0-45163816cc22">

### Suggested solution

I have two different proposals for this:
1. (Ideal) Can we make default workspaces names "<User's> Name"? Like "Isa's Workspace"
2. Alternatively, can we make them say "Your workspace" instead? I think that'd be better at least.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

**Comment by nickytonline** [comment]
I'd suggest `Your Workspace`. I can run a database migration for this. Something like this @jpmcb? I used your backfill for workspaces function as an example.

```sql
-------------------------------------------------------------------------------
-- This SQL file backfills personal workspaces for all users who do not yet
-- have a personal workspace.
--
-- This is only intended to be run once during the migration to personal workspaces.
-------------------------------------------------------------------------------
-- Declare a function to loop users and create personal workspaces for them
-- if they do not yet have a personal workspace
create
or replace function update_personal_workspace_names_for_users() returns void as $ $ declare user_record record;

begin for user_record in
select
    personal_workspace_id
from
    users
    and is_open_sauced_member = true loop -- generate a new workspace for the current user

update
    workspaces
set
    name = 'Your workspace',
    updated_at = now()
where
    id = personal_workspace_id;

end loop;

end;

$ $ language plpgsql;

-- Execute the function
select
    update_personal_workspace_names_for_users();
```

**Comment by jpmcb** [comment]
Yeah something like that but anything/everything we apply to prod we'll also want to add it to the `migrations/` as a record of change.

There's also more to be done than update current workspaces: we'll need to modify how new workspaces get created. Example:

https://github.com/open-sauced/api/blob/edd1a379e4d307c1df9f74cc1e7ed247e5e0d8cf/src/user/services/user.service.ts#L260-L263

Moving this issue to the API repo since it'll mostly encompass changes there.
